app/test/vesting_test.go

```diff
@@ -0,0 +1,550 @@
+		vestingtypes.Period{Length: int64(10), Amount: sdk.Coins{sdk.NewInt64Coin(app.BondDenom, 2500)}},
+	}
+
+	const accName = "period_vesting_0"
+	kr := testfactory.GenerateKeyring()
+	bacc, coins := testfactory.NewBaseAccount(kr, accName)
+	pva := vestingtypes.NewPeriodicVestingAccount(bacc, coins, startTime.Unix(), periods)
+
+	gsOpt := func(gs map[string]json.RawMessage) map[string]json.RawMessage {
+		vAccs := authtypes.GenesisAccounts{pva}
+		vBals := []banktypes.Balance{
+			{
+				Address: pva.GetAddress().String(),
+				Coins:   pva.GetOriginalVesting().Add(initBalanceForGasFee),
+			},
+		}
+
+		var err error
+		gs, err = AddAccountsToGenesisState(gs, vAccs...)
+		assert.NoError(s.T(), err)
+
+		gs, err = AddBalancesToGenesisState(gs, vBals...)
+		assert.NoError(s.T(), err)
+
+		return gs
+	}
+
+	cctx, _, _ := s.NewNetworkWithGenesisOpt(kr, gsOpt)
+
+	assert.NoError(s.T(), cctx.WaitForNextBlock())
+
+	accAddress := getAddress(accName, cctx.Keyring).String()
+
+	// Test account details correctness
+	cli := authtypes.NewQueryClient(cctx.GRPCClient)
+	res, err := cli.Account(context.Background(), &authtypes.QueryAccountRequest{
+		Address: accAddress,
+	})
+	assert.NoError(s.T(), err)
+
+	var qAcc vestingtypes.PeriodicVestingAccount
+	err = qAcc.Unmarshal(res.Account.Value)
+	assert.NoError(s.T(), err)
+
+	// Checking the queried account data
+	assert.Equal(s.T(), accAddress, qAcc.GetAddress().String(), "account address must match")
+	assert.EqualValues(s.T(), startTime.Unix(), qAcc.GetStartTime(), "start time must match")
+	assert.EqualValues(s.T(),
+		testfactory.BaseAccountDefaultBalance,
+		qAcc.GetOriginalVesting().AmountOf(app.BondDenom).Int64(),
+		"original vesting must match")
+	assert.EqualValues(s.T(), periods, qAcc.VestingPeriods, "periods must match")
+
+	/*--------*/
+
+	// Test available balance
+	for i := 0; i < 2; i++ { // We just let one period to be unlocked to keep some locked amount to test other stuff
+
+		currentLen := time.Since(startTime).Seconds()
+
+		balances, err := GetAccountSpendableBalance(cctx.GRPCClient, accAddress)
+		assert.NoError(s.T(), err)
+
+		expectedSpendableBal := initBalanceForGasFee.Amount.Int64()
+		passedLen := float64(0)
+		for _, pr := range periods {
+			passedLen += float64(pr.Length)
+			if currentLen > passedLen {
+				expectedSpendableBal += pr.GetAmount()[0].Amount.Int64()
+			}
+		}
+
+		assert.EqualValues(s.T(),
+			expectedSpendableBal,
+			balances.AmountOf(app.BondDenom).Int64(),
+			"spendable balance must match")
+
+		_, err = cctx.WaitForTimestamp(startTime.Add(periods[i].Duration() + 10*time.Millisecond)) // Wait for the next period to be passed
+		assert.NoError(s.T(), err)
+	}
+
+	s.testTransferVestingAmount(cctx, accName, false)
+	s.testDelegation(cctx, accName)
+	s.testClaimDelegationReward(cctx, accName)
+}
+
+func (s *VestingModuleTestSuite) TestGenesisContinuesVestingAccounts() {
```

Comment:
I think it would be helpful to include a godoc comment above this test that explains the behavior it is verifying.

[question] Perhaps `Continues` meant to be `Continuous`?